data-prepper-api/src/main/java/org/opensearch/dataprepper/model/plugin/PluginConfigPublisher.java

```diff
@@ -0,0 +1,17 @@
+package org.opensearch.dataprepper.model.plugin;
+
+/**
+ * An interface used to onboard and notify all {@link PluginConfigObservable} to invoke update.
+ * @since 2.5
+ */
+public interface PluginConfigPublisher {
+    /**
+     * Onboard a new {@link PluginConfigObservable}.
+     */
+    Boolean addPluginConfigObservable(PluginConfigObservable pluginConfigObservable);
```

Comment:
Returning `boolean` is probably best here so that we don't have a `null` possibility.